src/model/transaction/MosaicSupplyRevocationTransaction.ts

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2018 NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    AmountDto,
+    EmbeddedMosaicSupplyRevocationTransactionBuilder,
+    EmbeddedTransactionBuilder,
+    MosaicSupplyRevocationTransactionBuilder,
+    TimestampDto,
+    TransactionBuilder,
+    UnresolvedAddressDto,
+    UnresolvedMosaicBuilder,
+    UnresolvedMosaicIdDto,
+} from 'catbuffer-typescript';
+import { DtoMapping } from '../../core';
+import { Convert } from '../../core/format';
+import { UnresolvedMapping } from '../../core/utils/UnresolvedMapping';
+import { Address } from '../account';
+import { PublicAccount } from '../account/PublicAccount';
+import { UnresolvedAddress } from '../account/UnresolvedAddress';
+import { Mosaic } from '../mosaic';
+import { NetworkType } from '../network/NetworkType';
+import { Statement } from '../receipt';
+import { UInt64 } from '../UInt64';
+import { Deadline } from './Deadline';
+import { InnerTransaction } from './InnerTransaction';
+import { Transaction } from './Transaction';
+import { TransactionInfo } from './TransactionInfo';
+import { TransactionType } from './TransactionType';
+import { TransactionVersion } from './TransactionVersion';
+
+/**
+ * Creators of a revokable mosaic will be able to recall any and all balances from any holders. Holders of these mosaics implicitly place trust in the issuer.
+ * The mosaic issuer can revoke and recall balances using this transaction.
+ */
+export class MosaicSupplyRevocationTransaction extends Transaction {
+    /**
+     * Create a mosaic supply revocation transaction object
+     * @param deadline - The deadline to include the transaction.
+     * @param sourceAddress - Address from which tokens should be revoked.
+     * @param mosaic - Revoked mosaic and amount.
+     * @param networkType - The network type.
+     * @param maxFee - (Optional) Max fee defined by the sender
+     * @param signature - (Optional) Transaction signature
+     * @param signer - (Optional) Signer public account
+     * @returns {MosaicSupplyRevocationTransaction}
```

Comment:
the linter will complain I think. Should use description instead of types

Reply:
We don't have typedoc linter on new old ts SDK. Adding that should be a another PR that will affect many files